projects/templates/src/lib/components/po-page-dynamic-table/po-page-dynamic-table.component.ts

```diff
@@ -915,13 +915,20 @@ export class PoPageDynamicTableComponent extends PoPageDynamicListBaseComponent
     const metadata$ = this.getMetadata(serviceApiFromRoute, onLoad);
     const data$ = this.loadData();
 
-    const initialFilters = this.getInitialValuesFromFilter();
-
-    if (Object.keys(initialFilters).length) {
-      this.subscriptions.add(metadata$.subscribe());
-    } else {
-      this.subscriptions.add(concat(metadata$, data$).subscribe());
-    }
+    this.subscriptions.add(
```

Comment:
Sugestão:

Da forma proposta, não há necessidade de comentário sobre coverage, não precisamos declarar success e error, e não é executado mais um subscribe (caso manter precisamos adiciona-lo no this.subscriptions para remover depois)
```
   this.subscriptions.add(
      metadata$.pipe(
        switchMap(() => {
          const initialFilters = this.getInitialValuesFromFilter();

          if (!Object.keys(initialFilters).length) {
            return data$;
          }

          return EMPTY;
        })
      ).subscribe()
    );
  }
``` 

Para isso terá que ser alterado o teste unitário e no método **loadOptionsOnInitialize** retornar um of(null) em vez de EMPTY.

``` 
  private loadOptionsOnInitialize(onLoad: UrlOrPoCustomizationFunction) {
    if (onLoad) {
      return this.getPoDynamicPageOptions(onLoad).pipe(
        tap(responsePoOption =>
          this.poPageCustomizationService.changeOriginalOptionsToNewOptions(this, responsePoOption)
        )
      );
    }

    return of(null);
  }
```